crates/sui-types/src/storage.rs

```diff
@@ -969,3 +969,138 @@ impl Display for DeleteKind {
         }
     }
 }
+
+#[derive(Clone, Debug, Default)]
+pub struct OneShotSharedInMemoryStore(SharedInMemoryStore);
+
+impl OneShotSharedInMemoryStore {
+    pub fn insert_genesis_state(
+        &mut self,
+        checkpoint: VerifiedCheckpoint,
+        contents: VerifiedCheckpointContents,
+        committee: Committee,
+    ) {
+        let mut locked = self.0 .0.write().unwrap();
+        locked.insert_genesis_state(checkpoint, contents, committee);
+    }
+}
+
+impl ReadStore for OneShotSharedInMemoryStore {
```

Comment:
would be nice to avoid all the boilerplate code, but not a blocker.
We could use [delegate](https://docs.rs/delegate/latest/delegate/) crate or add a capacity parameter to original InMemoryStore(with default setting of u64::MAX)